api/src/services/schema.ts

```diff
@@ -0,0 +1,288 @@
+import { TYPES } from '@directus/shared/constants';
+import { Accountability } from '@directus/shared/types';
+import Joi from 'joi';
+import { Knex } from 'knex';
+import { version as currentDirectusVersion } from '../../package.json';
+import { ALIAS_TYPES } from '../constants';
+import getDatabase, { getDatabaseClient } from '../database';
+import { ForbiddenException, InvalidPayloadException } from '../exceptions';
+import { AbstractServiceOptions, Snapshot, SnapshotDiff, SnapshotWithHash } from '../types';
+import { applyDiff } from '../utils/apply-diff';
+import { getSnapshot, getVersionedHash } from '../utils/get-snapshot';
+import { getSnapshotDiff } from '../utils/get-snapshot-diff';
+
+const snapshotJoiSchema = Joi.object({
+	version: Joi.number().required(),
+	directus: Joi.string().required(),
+	vendor: Joi.string().valid('mysql', 'postgres', 'cockroachdb', 'sqlite', 'oracle', 'mssql', 'redshift').optional(),
```

Comment:
Should this Joi validation pull the list of vendors from the Database type definition instead? Pardon me if I am way off the mark, I am still getting acquainted with everything!

Reply:
Yes we should! I've just added type validation to the vendor list. Thanks! 👍